File: e-elearning-main/h-learning-fe-admin/src/dto/request/course.ts
export type CreateCourseReq = {
    metadata: {
        name: string
        description: string
        multiLogin: boolean
        value: number
        introduce: string
    }
    thumnail: File
}

export type UpdateCourseReq = {
    metadata: {
        id: number
        name?: string
        description?: string
        multiLogin?: boolean
        value?: number
        introduce?: string
    },
    thumnail: File
}

File: e-elearning-main/service/internal/dto/client/course.go
package requestdata

type CreateCourseReq struct {
	Name        string  `json:"name"`
	Description string  `json:"description"`
	MultiLogin  bool    `json:"multiLogin"`
	Value       float64 `json:"value"`
	Introduce   string  `json:"introduce"`
}

type UpdateCourseReq struct {
	Id          uint     `json:"id"`
	Name        *string  `json:"name"`
	Description *string  `json:"description"`
	MultiLogin  *bool    `json:"multiLogin"`
	Value       *float64 `json:"value"`
	Introduce   *string  `json:"introduce"`
}

type ChangeAvticeCourseReq struct {
	Id     uint `json:"id"`
	Active bool `json:"active"`
}
